%This is an example for a chapter, additional chapter can be added in the skeleton-thesis
%To generate the final document run latex, build and quick build commands on the skeleton-thesis file not this one.
%\chapter{Isolation by Environment}
\section*{Authors and Affiliations}
\begin{flushleft}
Ian J. Wang\textsuperscript{1},\\
Gideon S. Bradburd\textsuperscript{2},\\
\bf{1} Department of Environmental Science, Policy, and Management, University of California, Berkeley, CA
\\
\bf{2} Center for Population Biology, Department of Evolution and Ecology, University of California, Davis, CA, USA
\\
\bf{*} The authors contributed equally to this manuscript.
\end{flushleft}
%\section*{Abstract}
%The interactions between organisms and their environments can shape distributions of spatial genetic variation, resulting in patterns of isolation by environment (IBE) in which genetic and environmental distances are positively correlated, independent of geographic distance.  IBE represents one of the most important patterns that results from the ways in which landscape heterogeneity influences gene flow and population connectivity, but it has only recently been examined in studies of ecological and landscape genetics.  Nevertheless, the study of IBE presents valuable opportunities to investigate how spatial heterogeneity in ecological processes, agents of selection, and environmental variables contributes to genetic divergence in nature.  New and increasingly sophisticated studies of IBE in natural systems are poised to make significant contributions to our understanding of the role of ecology in genetic divergence and of modes of differentiation both within and between species.  Here, we describe the underlying ecological processes that can generate patterns of IBE, examine its implications for a wide variety of disciplines, and outline several areas of future research that can answer pressing questions about the ecological basis of genetic diversity.
\renewcommand{\thefigure}{2.\arabic{figure}}
\setcounter{figure}{0}
\renewcommand{\thetable}{2.\arabic{table}}
\setcounter{table}{0}

\section*{Introduction}
	Seventy-one years ago, Sewall Wright \citep{Wright1943} introduced the term ``isolation by distance" (IBD) to describe a pattern in which genetic differentiation at neutral loci increases with geographic distance.  The theory of isolation by distance describes the local accumulation of genetic differences when dispersal between populations or subgroups is geographically restricted \citep{Slatkin_1993}.  Effectively, neutral genetic differentiation between populations is the result of drift acting within populations more quickly than it is ameliorated by gene flow between populations \citep{Slatkin_1993, Rousset_1997}. Therefore, any processes that reduce the effective dispersal rate between populations will generate patterns of greater genetic differentiation \citep{Slatkin_1993,Bolnick_Otto_2013}.
	Decades of investigation into IBD have revealed it to be common in nature \citep{Slatkin_1993, Meirmans_2012} and brought a focus to the geography of population divergence and isolation \citep{Mayr_1963}.  However, geography represents only one of the key landscape components that can potentially influence gene flow and population connectivity \citep{Crispo_etal_2006, Lee_Mitchell-Olds_2011}.  Another important part of a landscape is the environment \citep{Nosil_etal_2005, Foll_Gaggiotti_2006, Thorpe_etal_2008}, and in the past decade, new fields like landscape genetics \citep{Storfer_etal_2007, Balkenhol_etal_2009, Wagner_Fortin_2013} have arisen to examine the roles played by ecology and the environment in microevolutionary processes \citep{McRae_Beier_2007, Räsänen_Hendry_2008, Bolnick_Otto_2013}.  One of the important concepts that has emerged is “isolation by environment” (IBE; \cite{Wang_Summers_2010}, which encapsulates the relationship between environmental heterogeneity and spatial variation in gene flow on a landscape \citep{Wang_etal_2013,Bradburd_etal_2013,Sexton_etal_2014}.
We define isolation by environment as a pattern in which genetic differentiation increases with environmental differences, independent of geographic distance, and which is agnostic with respect to the underlying processes that generated it.  The study of IBE comprises two challenges.  The first is to disentangle the relative strengths of IBD and IBE in observed patterns of spatial genetic differentiation, which can be a difficult statistical problem because geographic distance and environmental differences are often correlated (\ref{sfig:ibde}; \cite{Lee_Mitchell-Olds_2011, Wang_2013, Shafer_Wolf_2013, Bradburd_etal_2013}).  The second is then to determine the processes that have generated and maintained those patterns.  The development of new methods in spatial statistics and the rapid proliferation of both genomic and environmental GIS data have greatly facilitated research to overcome the first challenge of quantifying IBE relative to IBD, but researchers seeking to tease apart these patterns must still take care to employ appropriate study design, sampling strategies, and statistical techniques \citep{Bradburd_etal_2013,Sexton_etal_2014}.  The second challenge requires assessing the ways in which drift, selection, and dispersal have acted to shape patterns of genetic variation.  Quantifying the relative contributions of these processes, the ways in which they depend upon an ecologically heterogeneous landscape, and how those relative contributions vary across species or landscapes is an active and exciting field of research.
Here, we describe the processes that can generate this pattern and discuss important methods and considerations for studying IBE.  We conclude by proposing avenues for future research in IBE and suggesting a range of new and exciting questions about the ecological basis of spatial genetic variation.  The rise of IBE as a research focus has resulted in a tremendous opportunity to examine, often at very fine scales, the ways in which ecology shapes genetic variation in nature, forming a true bridge between the fields of population genetics and landscape ecology.

\begin{figure}[ht!]
\begin{center}
  \includegraphics[width=6in,height=2.5in]{figs/ibe/IBDE_triptych.pdf}
 \caption{Isolation by distance and environment.
 Under the patterns of isolation by distance (IBD) and isolation by environment (IBE), genetic distance increases with geographic and environmental distance.  The three panels show different views of a simulated dataset in which both patterns can be seen.  Points represent the genetic distance (Gen. Dist.) between a pair of populations, plotted against their geographic (Geo. Dist.) and environmental distances (Env. Dist.), and are heat-colored by the magnitude of that environmental distance.
 \label{sfig:ibde}}
\end{center}
\end{figure}

\section*{Definition of IBE}
Isolation by environment is defined as a pattern in which genetic differentiation increases with environmental differences, independent of geographic distance (\ref{sfig:ibde}; \cite{Wang_Summers_2010, Bradburd_et al_2013; Sexton_etal_2014}).  The important environmental variables may be continuous, such as elevation or humidity \citep{Murphy_etal_2010, Bradburd_etal_2013}, or discrete, such as habitat or substrate type \citep{Andrew_etal_2012}.  They may describe abiotic factors, like temperature and precipitation \citep{Wang_2012}, or biotic factors, like vegetation density and host \citep{Via_Hawthorne_2002}.  The key is that they contribute to explaining variation in pairwise genetic distances beyond that explained by geographic distance (and after accounting for any correlation between environmental and geographic distances).  This definition is solely a description of a pattern and is agnostic with respect to the processes that have generated that pattern.  
We advocate for this pattern-based, rather than process-based, definition because there are frequently many biological processes that can generate a given pattern of observed genetic differentiation.  This definition provides the general case for which numerous other “isolation by” terms present specific, process-based instances.  For example, isolation by adaptation (IBA), defined as the correlation of adaptive phenotypic and neutral genetic divergence \citep{Nosil_2008}, and isolation by ecology, defined as the correlation of ecological and neutral genetic divergence \citep{Claremont_etal_2011, Edelaar_etal_2012}, both describe patterns due to a specific selective mechanism.  These process-based terms are designed to address the important question of how observed patterns of spatial genetic variation are generated; however, the processes that have generated these empirical patterns of divergence cannot be observed directly from these patterns, and the same empirical pattern of genetic differentiation could be due to many different underlying processes.
Our definition can also be contrasted with isolation by resistance (IBR), defined as the correlation of genetic and ``resistance" distances \citep{McRae_Beier_2007}.  Resistance distance between a pair of populations can be understood as the probability that an individual disperses from one to the other, integrating over all paths that individual might take, and weighting those paths by their “friction” to dispersal (a low pairwise resistance means a high probability of dispersal, and vice versa).  Isolation by resistance, like our definition of IBE, is also agnostic with respect to process; the dispersal resistance of a specific landscape element may be due to selection against maladapted dispersers, migratory preference, or simple cost of transport.  However, IBR implicitly conflates IBD and IBE, making it impossible to differentiate the strengths of these two patterns in empirical data.
Our simple but broad definition is meant to be inclusive of all positive associations between genetic and environmental distances, in keeping with the tradition set up by the definition of IBD from \citet{Wright_1943}.  It provides a collective term for a genetic pattern that is common in nature but may be generated by many different processes, and it implicitly links environmental variation on a landscape to gene flow and population structure.  Thus, IBE is valuable for understanding spatial genetic differentiation in the context of landscape variation.

\section*{Processes Generating IBE}
Isolation by environment is a pattern that can be generated by a variety of ecological processes.  These may be relatively simple, like when a temperature cline regulates dispersal among populations of an ectotherm \citep[e.g.][]{Murphy_etal_2010}, or they may represent more complex ecological interactions, like when genetic differentiation between plant populations is mediated by differences in their pollinator communities \citep[e.g.][]{Hopkins_etal_2012}.  This many-to-one mapping of process to pattern can make it difficult, though by no means impossible, to learn about the mechanisms generating observed patterns of spatial genetic variation.  The first step for these investigations, and an important component for any study of IBE, is to carefully consider the potential processes that could be taking place.  Below, we identify four ecological processes – not mutually exclusive – that can generate a pattern of IBE, including (1) natural selection against immigrants, (2) sexual selection against immigrants, (3) reduced hybrid fitness, and (4) biased dispersal \ref{sfig:ibe_causes}.

\begin{figure}[ht!]
\begin{center}
  \includegraphics[width=3.5in,height=3.5in]{figs/ibe/IBE_causes.pdf}
 \caption{Illustration of processes that can generate a pattern of isolation by environment.
 Dispersal between divergent environments can be reduced when (1) natural selection acts upon immigrants adapted to different environmental conditions, (2) sexual selection limits the reproductive success of immigrants with alternative traits, (3) hybrid offspring of native and immigrant parents have reduced fitness, for instance due to intermediate phenotypes, (4a) biased dispersal resulting from a genotype or phenotype leading to a dispersal preferences for particular environments, or (4b) biased dispersal resulting from a plastic natal habitat preference.
 \label{sfig:ibe_causes}}
\end{center}
\end{figure}


\subsection*{(1) Natural Selection Against Immigrants}
Natural selection can generate IBE among populations inhabiting different environments when these populations are locally adapted \citep{Nosil_etal_2005, Räsänen_Hendry_2008}.  In these cases, populations evolve traits suited to their local environments, regardless of their fitness consequences in other environments \citep{Servedio_2004, Kawecki_Ebert_2004, Nosil_etal_2005}.  Thus, native genotypes in each environment will have, on average, higher fitness than immigrant genotypes originating in different environments \citep{Servedio_2004, Kawecki_Ebert_2004}.  When individuals or populations show ecological specialization \citep{Lu_Bernatchez_1999, Via_Hawthorne_2002}, divergent natural selection will limit the reproductive success of individuals moving into different environments from which they are adapted \citep{Räsänen_Hendry_2008, Mosca_etal_2012}.  For instance, walking sticks adapted to appear cryptic on certain host plants experience greater predation from visually-oriented predators after moving to a different host species \citep{Nosil_2004, Nosil_etal_2005}, and divergent selection regimes in inland habitat with seasonal drought and coastal habitat with year round soil moisture cause nearly complete reductions in gene flow between populations of yellow monkey flowers adapted to either climate \citep{Lowry_etal_2008}.  We can expect the strength of divergent selection to be proportional to the magnitude of the differences among environments \citep{Crispo_etal_2006, Lee_Mitchell-Olds_2011, Wang_etal_2013}, and therefore, pairs of populations inhabiting increasingly different environments will experience reduced gene flow and greater genetic divergence \citep{Nosil_etal_2005, Thorpe_etal_2008}.  This environmentally associated natural selection may generate either pre- or post-mating reproductive isolation, depending on whether immigrants survive and thrive long enough to mate locally. If immigrants are able to mate locally, there may subsequently be selection against immigrant alleles in hybrids (see ``Reduced Hybrid Fitness" below). 

\subsection*{(2) Sexual Selection Against Immigrants}
Similarly, divergent sexual selection among populations inhabiting different environments can also generate IBE \citep{Servedio_2004; Nosil_etal_2005; Safran_etal_2013}.  When populations inhabiting different environments show divergence in mate choice or sexual signals, sexual selection will reduce the reproductive success of dispersers moving between them \citep{Servedio_2004; Nosil_etal_2005}.  In some cases, divergent sexual selection will be related to environmentally driven natural selection.  For instance, under the good genes hypothesis, mate choice evolves so that individuals prefer mates possessing traits that increase offspring fitness \citep{Ingleby_etal_2010}, and if the fitness conferred by these traits varies across environments, divergent sexual selection will result and lead to variation in preferences for ecologically important traits \citep{Nosil_etal_2005}.  This is the case with lesser wax moths, in which females choose males with different signals to produce offspring that mature faster in different food and temperature environments \citep{Jia_Greenfield_1997}.  In other cases, divergent sexual selection can also be related to environmental variation but not natural selection.  For instance, under the sensory drive hypothesis, sexual signals and their perception by receivers can evolve to be more effective under local environmental conditions, and therefore, dispersers may have reduced reproductive opportunities if their sexual signals are viewed in a different ecological context.  Such is the case with some cichlids \citep{Seehausen_etal_2008} and sticklebacks \citep{Boughman_2001}, in which adaptive variation in visual sensitivity leads to genetic isolation between groups expressing different nuptial coloration across gradients in ambient light conditions.  Thus, divergent sexual selection can act in concert with divergent natural selection or independently of it \citep{Servedio_2004; Nosil_etal_2005; Safran_etal_2013}, in both cases producing increased levels of IBE.  As with natural selection against immigrants, imperfect sexual selection against non-local individuals can allow for the creation of hybrid offspring, the sexual characteristics and mating behavior of which may themselves be selected against (see ``Reduced Hybrid Fitness" below).

\subsection*{(3) Reduced Hybrid Fitness}
Selection can further contribute to IBE when hybrid offspring of immigrant-native parent crosses have reduced fitness relative to their non-hybrid neighbors \citep[i.e. post-zygotic extrinsic reproductive isolation of ecologically divergent populations;][]{Nosil_etal_2005; McBride_Singer_2010}.  If hybrid offspring have intermediate phenotypes, then they may not occupy an available ecological niche in their natal environment or may have limited mating opportunities, both of which will reduce effective rates of long-term gene flow \citep{Nosil_etal_2005, Garant_etal_2007, McBride_Singer_2010}.  For instance, in Euphydryas butterflies, hybrids from parents adapted to different hosts exhibit intermediate traits that are significantly maladaptive, including foraging and oviposition behaviors \citep{McBride_Singer_2010}.  These cases will mostly serve to strengthen the patterns resulting from natural or sexual selection on immigrants.  However, we consider this separate from these processes, as other authors have \citep[e.g.][]{Servedio_2004, Nosil_etal_2005}, because selection acts on a different generation of individuals, rather than on the immigrants themselves, and under certain scenarios, the determinants of offspring fitness may be different from those of immigrant fitness.  For example, if combinations of alleles that arose in isolated parental populations are incompatible, hybrid fitness may be reduced due to intrinsic, rather than extrinsic reproductive isolation \citep[e.g. Dobzhansky-Muller Incompatibilities; ][]{Dobzhansky_1937}; in this case, selection would not be against immigrant alleles, but rather against the combination of immigrant and native alleles.

\subsection*{(4) Biased Dispersal}
Isolation by environment can also result when a genotype, phenotype, or behavior contributes to a dispersal preference for a particular environment.  Under these cases of 'biased' or 'directed' dispersal, an individual's traits affect the likelihood of moving to various habitats.  This may be due to heritable variation \citep{Edelaar_etal_2008, Bolnick_Otto_2013} or may be the result of a behavioral or plastic response induced by an individual’s natal or developmental environment \citep{Davis_Stamps_2004}.  A dispersal bias may arise because of a fitness or performance advantage in a particular environment, like the matching of coloration and habitat in white sand lizards \citep{Rosenblum_Harmon_2011}.  In this case, the resulting pattern of dispersal naturally leads to the correlation of genotype and environment \citep{Bolnick_Otto_2013}.  However, such a pattern may also arise even when individuals do not necessarily experience differential selection in different habitats, for instance when dispersers avoid novel habitat when moving across an environmentally heterogeneous landscape \citep{Stevens_etal_2005, Feder_Forbes_2007} or when individuals prefer to disperse to habitats similar to their native habitat, known as natal habitat preference induction \citep{Davis_Stamps_2004}.  This type of environmentally-induced plastic response can be seen in two species of true frogs, in which the exposure of eggs to olfactory cues in water led to a preference among tadpoles for those environmental cues that is maintained through metamorphosis \citep{Hepper_Waldman_1992}.  In such cases, there does not need to be local adaptation, but a pattern of IBE may still arise.  We consider these processes distinct from natural or sexual selection against immigrants because selection may not actually act upon these individuals – there are no deaths or reproductive consequences since individuals move before any selective events can occur \citep{Bolnick_Otto_2013}.

\section*{Methods and Considerations for Studying IBE}
\subsection*{Sampling Scheme}
When investigating patterns of spatial genetic variation, a researcher must design a sampling scheme that allows the study to disentangle the relative effects of geographic distance and ecological or environmental distance.  This includes two important considerations: sampling to maximize the range of observed geographic and ecological distances, and sampling to minimize the correlation between the potential explanatory variables.  By sampling over an extensive range of geographic and environmental distances, the researcher can learn about the shape of the decay of genome-wide relatedness with different distances.  For example, if patterns of IBD or IBE increase non-linearly, a researcher with a sampling scheme that includes only small distances and large distances might be unable to quantify well the shape of that curve (\ref{sfig:sampling}).
	Additionally, researchers must avoid a sampling scheme in which ecological and geographic distance are confounded.  For example, in the confounded sampling configuration shown in \ref{sfig:sampling}, significant geographic distance is strongly correlated with environmental distance, making it difficult to statistically disentangle their effects.  Instead, samples from different environments should be drawn from a mixture of populations that are geographically close and distant, and samples from within similar environments should be as well.  This will serve to reduce the association between geographic and environmental distances.  
	For both considerations – maximizing the range of sampled pairwise distances and minimizing the correlation of covariates – a limited sampling scheme could lead to uninterpretable or misleading results.  However, empirical researchers are frequently limited in their sampling effort, and are therefore forced to make compromises in the number of locations or individuals they are able collect and genotype.  When designing sampling schemes with these compromises in mind, we suggest first examining simple histograms of pairwise distances (geographic and environmental) and correlations between environmental and geographic distances under a projected sampling scheme prior to sample collection.  If necessary, adjustments to the proposed scheme can then be made to increase the diversity of pairwise distances sampled or to minimize the correlation between these variables.
	Above, we have placed an emphasis on placement and number of sampled locations, the latter of which is negatively correlated with the number of individuals per sampling location that a researcher can afford to genotype.  Increasing the number of individuals sampled per location decreases the binomial sampling noise around estimates of population allele frequencies and is therefore useful in accurately estimating pairwise population differentiation.  However, in the era of next-generation sequencing, it is possible to sequence each individual at large numbers of loci, which, assuming they are independent, serve as independent instantiations of the coalescent process.  With many sequenced loci in two populations, it is therefore possible to get good estimates of the differentiation between them, even if the sample size at each locus is small, and the estimate of differentiation is therefore noisy \citep{Patterson_etal_2006}.
In summary, researchers designing studies to investigate IBE should seek to maximize the range of geographic and environmental distances between sampled locations, minimize the correlations of those geographic and environmental covariates, and maximize the number of sampled locations even at the cost of sample size.  In all cases, the relevant question that should be kept in mind is not whether it will be possible to detect spatial genetic differentiation but whether it will be possible to distinguish a genetic pattern of IBE from the background pattern of IBD.

\begin{figure}[ht!]
\begin{center}
  \includegraphics[width=4in,height=4.5in]{figs/ibe/sampling_scheme_figure.pdf}
  %figs/ibe/sampling_scheme_figure_hires.png}
 \caption{Illustration of common pitfalls in designing a sampling scheme.
These panels show the contrasted outcomes between complete sampling and an inadequately designed sampling scheme for populations on a hypothetical landscape with environmental variation (A).  The histogram of pairwise distances (B) obtained by a poor sample design shows that this design results in obtaining pairwise distances that are not fully representative of the full set of populations.  The comparison of the environmental and geographic distances recovered under the complete and partial sampling schemes (C) shows that these distances are highly correlated under the partial sampling scheme; IBD and IBE are conflated because all comparisons over short distances are also similar in environment, and all comparisons over long distances are environmentally disparate, with no intermediate comparisons to help disentangle the relative contributions of IBD and IBE.  Finally, pairwise genetic differentiation plotted against pairwise geographic distance and colored by environmental distance (D) shows that under the partial sampling scheme, the shape of the curves for IBD and IBE are poorly estimated.
 \label{sfig:sampling}
  }
\end{center}
\end{figure}


\subsection*{Statistical Techniques}
This same question should be at the heart of the statistical techniques used to study IBE; researchers must account for IBD when attempting to quantify IBE along one or more environmental axes.  This task is inherently difficult because measures of pairwise genetic distance are non-independent, and therefore should not be modeled directly as a function of pairwise geographic or ecological distances.  The Mantel and partial Mantel tests were designed to avoid this problem of non-independence by implementing a significance test that explicitly accounts for the pairwise nature of the dependent and independent variables.  However, when the data are spatially autocorrelated, the partial Mantel has a pathological type I error rate \citep{Guillot_Rousset_2013}, as the significance procedure implicitly rejects the spatial structure of the data.
However, the question the partial Mantel was designed to answer in landscape genetics is still vital to many research programs today: what are the relative contributions to observed patterns of spatial genetic variation from geographic and ecological distances?  Recently, several new, more statistically robust methods designed to answer this question have been released.  These include methods for examining genome-wide patterns of differentiation and for investigating divergence in individual loci, both of which are important for understanding the nature of IBE.  Coupled with new techniques for treating the spatial component of genetic variation more explicitly, these provide strength and flexibility for studying IBE in almost any natural system. 
New methods for examining general patterns of genetic differentiation fall into two categories: modeling covariance in allele frequencies and matrix regression approaches.  Of the former, BEDASSLE \citep{Bradburd_etal_2013} is a Bayesian method that models the covariance in allele frequencies across the genome as a decreasing function of pairwise geographic and ecological distances.  The coefficients estimated for these pairwise distance elements can be compared to quantify the relative impact of geographic and ecological distances on patterns of genetic variation.  This method was applied to a maize dataset and revealed that 1000m of elevation difference between populations has the same effect on genetic differentiation as about 150km of horizontal distance \citep{Bradburd_etal_2013}.  Of the latter, methods based on matrix regression, including generalized dissimilarity modeling \citep[GDM;][]{Freedman_etal_2010}, structural equation modeling \citep[SEM;][]{Wang_etal_2013}, and multiple matrix regression with randomization \citep[MMRR;][]{Wang_2013} apply a regression framework to simultaneously quantify the effects of multiple distance matrices on a single response variable, typically genetic distance, using different computational methodologies.  SEM was used to infer that IBD typically contributed about twice as much to genetic divergence as IBE in 17 species of anoles and to quantify the relative contributions of individual environmental variables to IBE in those species \citep{Wang_etal_2013}.  Both sets of methods have the power to disentangle IBD and IBE, and the preferred method to be used will depend upon the data available and the question to be answered.
New techniques for dealing with genetic distances in a spatially explicit manner include those that handle continuous spatial variation and those designed to account for the spatial arrangement of population networks.  For continuous variation, LocalDiff \citep{Duforet-Frebourg_Blum_2014} uses Bayesian kriging and MEMGENE uses Moran’s eigenvector maps \citep{Galpern_etal_2014} to detect discontinuities in patterns of gene flow across a landscape, which may be associated with landscape heterogeneity and barriers to dispersal.  For population networks, the utilization of conditional genetic distances derived from population network topology can improve estimation of IBD and IBE, and can be used with phylogeographic history to separate the effects of historical and contemporary barriers to gene flow \citep{Dyer_etal_2010}.  This method was used to parse out the effects of phylogeographic history in the Sonoran desert succulent, revealing individual effects of spatial and bioclimatic variables on genetic differentiation \citep{Dyer_etal_2010}.  Both of these techniques are valuable for handling spatial genetic distance data, especially when paired with an appropriate statistical technique for disentangling IBD and IBE. 
These approaches all consider genome-wide patterns of differentiation, and are therefore designed to answer a separate set of questions from methods like Bayenv2 \citep{Coop_etal_2010, Günther_Coop_2013}, which asks whether individual SNPs are correlated with environmental variation, and spaMM \citep{Rousset_Ferdy_2014}, which is designed to account for spatial autocorrelation in the association of a genotype with some environmental variable.  The former was applied to an Atlantic herring dataset to identify loci that were strongly differentiated along a salinity gradient \citep{Günther_Coop_2013}.  Once patterns of differentiation have been identified, additional methods can be applied to genomic data to attempt to elucidate the population-level processes that have generated those patterns.  These methods, which are principally designed to look for the signature of selection across the whole genome, include identifying signals of selective sweeps and genomic hitchhiking \citep{Via_2012, Flaxman_etal_2013} and quantifying patterns of introgression in different regions of the genome characterized by different recombination rates \citep[e.g.][]{Geraldes_etal_2011}.  All of these methods represent significant steps forward in the statistical analysis of patterns of spatial genetic variation and IBE; as next-generation sequencing power enables researchers to collect genomic data on ecological scales (many samples over broad geographic sampling areas), we expect both their use and their utility to increase.

\subsection*{Experimental Data}
Statistical techniques are valuable for quantifying IBE, but an experimental approach remains the best way to establish a causal mechanism.  By manipulating organisms and the environments in which they occur, it is possible to distinguish between different processes that could produce IBE.  A detailed review of the full range of experimental procedures used to learn about biological processes generating genetic variation is beyond the scope of this paper.  However, below, we highlight a selection of the most commonly used experimental tools, and discuss the ways in which they may be used to learn about the processes generating a pattern of IBE.  
To determine if natural selection against immigrants is driving a pattern of IBE, researchers can use reciprocal transplant experiments, multiple common gardens, or provenance tests \citep{Thorpe_etal_2005, Leinonen_etal_2011}.  A fitness advantage observed when organisms are matched with their local environment is evidence for local adaptation and natural selection against immigrants.  For instance, reciprocal transplant experiments on monkey flowers demonstrated natural selection against immigrants adapted to different soil moisture \citep{Lowry_etal_2008}.
To determine if sexual selection has generated IBE, researchers can use mate choice trials.  Individuals from the same environment mating assortatively when presented with options from similar and different environments can be taken as evidence that sexual selection is acting to shape patterns of genetic variation between populations, as was the case with laboratory matings of cichlids that showed assortative mating between geographically close populations that differed in male nuptial coloration \citep{Knight_Turner_2004}.  
Researchers studying organisms with sufficiently short generation times and tractable reproductive behavior can create hybrids between populations from ecologically divergent habitats and assess their fitness in a common garden or in both parental environments to test for reduced hybrid fitness generating IBE.  For instance, randomized common garden experiments on hybrid \textit{Arabidopsis lyrata} from ecologically divergent populations with differences in timing of flowering and floral display traits demonstrated variation in the fitness of hybrids relative to parent populations \citep{Leinonen_etal_2011}. 
Researchers should also consider the possibility that non-selective processes, such as biased dispersal or differentially resistant landscape elements, have led to decreased gene flow between populations. These possibilities can be experimentally examined using controlled or reciprocal release experiments \citep[e.g.][]{Bolnick_etal_2009}, radio-tracking experiments \citep[e.g.][]{Broquet_etal_2006}, stable isotope analysis \citep[e.g.][]{Pilot_etal_2012}, or experimental quantification of an organism’s dispersal ability in different environments \citep[e.g.][]{Stevens_etal_2005}.  In a controlled or reciprocal release, patterns of dispersal that are biased toward native habitat are evidence that observed IBE is due to biased dispersal.  For example, in lake and stream sticklebacks, mark-transplant-recapture experiments showed that a large majority returned to their native habitat and that dispersal into nonnative habitat was phenotype dependent \citep{Bolnick_etal_2009}.  Radio-tracking and stable isotope analysis, which can reveal biased patterns of organisms moving over and utilizing different habitats within a landscape, can also provide strong evidence that biased dispersal is generating IBE.  For instance, stable isotope profiles have been used to reveal correlations between diet differences, associated with habitat choice, and genetic distances in European wolves \citep{Pilot_etal_2012}.  Finally, quantification of an organisms’ ability to disperse over or between different substrates can also be indicators of biased dispersal.  For instance, mobility analysis in an experiment arena composed of different habitats provided quantitative estimates of dispersal ability over different substrates in the natterjack toad \citep{Stevens_etal_2005}.  
	Finally, because these processes are not mutually exclusive, multiple processes may act or have acted to generate the observed pattern of IBE.  For example, with reinforcement, the presence of reduced hybrid fitness is expected in the long run to select for increased pre-mating sexual reproductive isolation between parental populations.  Therefore, it may be necessary to follow multiple lines of experimental evidence to determine the ecological processes that have generated patterns of spatial genetic differentiation.

\subsection*{Caveats}
There are a number of factors that can potentially confound the detection and measurement of IBE.  Population history, demography, and heterogeneity may all influence estimates of observed IBE and potentially lead to migration-drift disequilibrium. Sampling design can also compound the challenges posed by these factors, potentially leading to inferential errors.  If, for example, some sampled populations are only recently diverged, and not in migration-drift equilibrium, the estimated rates of IBD and IBE in the complete set of populations may be biased \citep{Marko_Hart_2011}.  In addition, model inadequacy poses a serious problem in the analysis of landscape genetic data, in which the processes that have generated the data are almost sure to be vastly more complex and idiosyncratic than the models used to perform inference.  Researchers are well advised to assess model adequacy, either through evaluating model fit, or by performing an explicit test of adequacy, such as posterior predictive simulation \citep[e.g.][]{Bradburd_etal_2013} and to be cautious in the interpretation of their results.  Recognizing these potentially confounding factors and properly designing a study to account for them is critical for accurately detecting IBE.

\section*{Broader Implications}
The study of IBE and the mechanisms generating it have significant implications for a variety of disciplines.  The most obvious example is landscape genetics - much of landscape genetics has focused on examining how landscapes influence population connectivity \citep{Storfer_etal_2007, Sork_Waits_2010, Wagner_Fortin_2013}, and analysis of IBE is clearly important for fully understanding how landscape and environmental features influence gene flow and population structure \citep{Wang_etal_2013, Bradburd_etal_2013, Sexton_etal_2014}.  This can add a crucial element for assessing the importance of different parts of the landscape for corridor and reserve design and for performing long-term population viability analyses, which are clearly valuable for conservation efforts.  For species in which IBE is prominent, conservationists might also have to consider whether IBE results from local adaptation and whether to prioritize the protection of gene flow between more similar habitats over gene flow from divergent habitats that could 'swamp out' locally adaptive genotypes \citep{Aitken_Whitlock_2013}.
	Along these lines, IBE can also contribute to studying the ecology of local adaptation, currently a topic of major interest \citep{Aitken_Whitlock_2013, Blanquart_etal_2013, Günther_Coop_2013, Butlin_etal_2014}, potentially providing an efficient first step in identifying systems with adaptive divergence.  However, because many mechanisms can generate IBE, detection of this pattern alone is not evidence of local adaptation.  Similarly, under the simplifying assumption that IBE always results from selection, some previous studies have interpreted IBE as evidence for incipient ecological speciation \citep{Shafer_Wolf_2013}.  By definition, ecological speciation is the evolution of reproductive isolating barriers due to divergent selection under different ecological conditions \citep{Lu_Bernatchez_1999, Schluter_2009, Thibert-Plante_Hendry_2010}.  Because IBE can result from processes other than selection, it should not be taken at face value as evidence for ecological speciation; a more prudent interpretation is that IBE can indicate that the underlying conditions necessary for ecological speciation exist in a given system.  Nevertheless, IBE is still valuable for studying local adaptation and ecological speciation because many methods can evaluate which environmental variables contribute to IBE, and this can aid in designing experiments to identify the ecological factors driving adaptive population divergence and isolation.  Moreover, because of its association with gene flow, IBE is also valuable for examining the important question of how local adaptation occurs in the face of ongoing gene flow \citep{Saint-Laurent_etal_2003, Nosil_Crespi_2004, Räsänen_Hendry_2008, Muir_etal_2014, Butlin_etal_2014}.  In these scenarios, the pattern of gene flow may be more important than the level of overall gene flow - for instance if gene flow is primarily among similar environments - and studies that contrast IBE with IBD could contribute to our understanding of the types of gene flow that enable adaptive population divergence.
	Finally, studies of IBE can also influence landscape and community ecology.  For instance, the general prevalence of IBE in nature \citep{Shafer_Wolf_2013, Sexton_etal_2014} has ramifications for traditional ecological theory about the distribution of individuals in space, like the ideal free distribution, which posits that organisms are free to move among habitat patches unimpeded and will distribute themselves among them in proportion to the availability of resources.  In general, a pattern of IBE means that these assumptions are unmet, as individuals will either have a non-random distribution among patches driven by factors aside from resource availability \citep{Nosil_etal_2005, Bolnick_Otto_2013} or will be unable to move freely among patches because of some spatial ecological process mediating dispersal \citep{Lu_Bernatchez_1999, Crispo_etal_2006, Räsänen_Hendry_2008, Bradburd_etal_2013}.  Additionally, if IBE is present across many species within a community, then they may show spatially similar patterns of divergence among populations, and this could potentially lead to the co-diversification of multiple species \citep{Johnson_Stinchcombe_2007}.  Thus, IBE, its prevalence, and its underlying factors can have significant implications for a wide variety of ecological and evolutionary processes.

\section*{Future Directions}
Despite the growing interest in IBE, many exciting areas remain open for future research \citep{Balkenhol_etal_2009, Storfer_etal_2010}.  Here, we outline five areas of pressing interest that present a wealth of opportunities for innovative research in the near future:  (1) landscape genomics, (2) comparative landscape genetics, (3) population heterogeneity, (4) temporal variation, and (5) identifying the underlying ecological processes that drive IBE.  Investigating these areas and answering the important questions they present will greatly expand our knowledge of how ecology influences genetic variation across space, time, taxa, and the genome.

\subsection*{(1) Landscape Genomics}
Integrating population genomics into landscape ecological research is an exciting frontier for landscape genetics that will open up many new avenues of scientific inquiry.  Several recent studies have already explored how landscape genomics can provide greater power and resolution for examining spatial patterns and adaptive variation \citep[e.g.][]{Coop_etal_2010, Parchman_etal_2012, Lasky_etal_2012, Vincent_etal_2013, Yoder_etal_2014}.  However, one question that has not yet been extensively investigated is how the environment influences variation differentially across the genome.
	Different sites across the genome can experience different evolutionary scenarios because of the dynamics of the many processes that act on genetic variation \citep{Nosil_etal_2008, Turner_Hahn_2010, Flaxman_etal_2013, Soria-Carrasco_etal_2014}.  While the neutral process of drift due to decreased gene flow between a pair of populations acts on the entire genome, selective forces, like natural and sexual selection against immigrants, will only target the loci involved in the traits under selection \citep{Nosil_etal_2008, Turner_Hahn_2010}.  A pattern of IBE due to selective forces may therefore be observed locally at a given locus but not be seen globally across the entire genome, and the localization of these effects will depend partly on the rate of recombination, which itself may vary across the genome \citep{Nosil_etal_2008}.  Recent advances in population genetics have significantly improved the inference of heterogeneous coalescent histories across the genome \citep{Ralph_Coop_2013, Harris_Nielsen_2013}, and we think that the incorporation of ecological processes into these methods will be an exciting way forward.  This could reveal how various signatures of IBE due to different underlying environmental factors are expressed across the genome – with regards to the localization and relative strengths of IBD and IBE across gene regions with different functions, architecture, and recombination rates – and such studies could yield unprecedented looks into the ecological factors that drive genetic divergence in nature. 

\subsection*{(2) Comparative Landscape Genetics}
	Studies of single species on single landscapes are undoubtedly valuable for landscape genetics, often providing important information on organisms that are ecologically interesting or of conservation concern.  However, comparative studies, either of multiple species or multiple landscapes, are likely to provide the next big advances for understanding organism-landscape interactions.  These comparisons can reveal the factors that drive IBE, generally, and whether they are intrinsic to organisms or to landscapes.  Examinations of multiple species on one landscape \citep[e.g.][]{Goldberg_Waits_2010, Richardson_2012} can answer whether multiple taxa are affected in similar ways by a particular landscape, and studies of single species across multiple landscapes \citep[e.g.][]{ShortBull_etal_2011, Trumbo_etal_2013} can answer whether the influences of landscapes on organisms are the result of the specific spatial structure of the landscape or the biology of the organisms themselves.  These studies may be of particular interest when trophic relationships between study organisms are known and provide a set of predictions about their relative patterns of population structure (e.g. patterns of structure in predator-prey or host-parasite systems), but all such studies will help in understanding how organism-landscape interactions contribute to IBE.

\subsection*{(3) Population Heterogeneity}
To date, most studies have performed their analyses at the species or meta-population level, assuming that ecological responses are essentially constant across all populations.  However, populations may diverge in ways that are important for how they interact with the landscape, including dispersal ability, habitat preference, and adaptation to different ecological conditions, all of which may influence patterns of gene flow.  Spatial variation in evolutionary processes, like selection, can also affect gene flow and lead to different factors driving IBE in different populations.  So, despite a focus on spatial variation, few landscape genetics studies have considered how spatial variation in the traits and factors affecting gene flow contribute to estimates of IBE and its causal factors at higher levels.  If populations vary substantially in their disposition towards IBD and IBE, then the subset of populations chosen for an empirical study may also influence the overall estimation of landscape effects on a given species.  In any case, new studies that explicitly consider this possibility will provide insight into how spatial variation among populations shapes IBE.

\subsection*{(4) Temporal Variation}
Another important consideration for studies of IBE is temporal variation, especially because evolutionary forces, ecological processes, and environments typically change through time.  While spatial variation and scale have been identified as important factors in landscape genetic analysis \citep{Cushman_Landguth_2010}, the implications of temporal variation have not yet been thoroughly investigated, although some recent studies provide indications of their importance \citep{Crispo_Chapman_2009, Dyer_etal_2010, Epps_etal_2013, He_etal_2013}.  Ideally, studies would be conducted on temporally spaced samples and geospatial data from corresponding time periods, potentially including museum specimens \citep{Nachman_2013}, but new analytical methods may also allow inferences of historical patterns \citep{Dyer_etal_2010, He_etal_2013}.  In either case, these studies should provide valuable insights into the tempo of change in ecological drivers of diversification and the relative importance of historical landscape factors for explaining contemporary patterns of variation.  This is particularly important for understanding how species will respond to changing climate and environmental conditions, a commonly stated goal of landscape genetic studies.  Probably the best way to predict how species will be affected by a changing environment is to account for how they responded to changing conditions before.

\subsection*{(5) Identifying Underlying Ecological Processes}
	Finally, while the identification of IBD, IBE, and related patterns still presents many interesting research objectives, we believe that it is important for future research to go beyond describing patterns of spatial variation and to begin testing for the underlying mechanisms that generate these patterns.  One of the fundamental charges of ecology and evolutionary biology is to explain the patterns of variation – both phenotypic and genetic – that we observe in nature.  The first step is quantifying the observed patterns, and the second is investigating the processes that generate them   Sometimes this may mean that research on IBE will have to extend beyond landscape genetics techniques and use experimental and traditional landscape ecological methods, like reciprocal transplants, common gardens, radio tracking, functional morphology, and biomechanics, to examine whether divergent populations exhibit differences consistent with a particular mode of differentiation.  Targeted population genetic analyses, applied with proper sampling designs, can also be valuable for examining spatial variation in signals of selection and the dispersion of genotypes linked to ecologically important traits across environmental gradients.  These studies will significantly expand the scope of investigations into IBE and should provide remarkable new insights into the causal ecological factors driving genetic variation in the wild.

\section*{Conclusions}
	Over the last decade, landscape genetics has made remarkable progress in examining the impacts of landscape variation on gene flow and population dynamics \citep{Sork_Waits_2010, Storfer_etal_2010}.  Isolation by environment will play a major part in the next big steps, providing a framework for examining how ecological and environmental heterogeneity shape the distribution of genetic variation in nature.  The environment is clearly a core component of the landscape – even though it has not been explicitly considered in landscape genetics until recently – which can significantly influence gene flow and population connectivity.  As suggested by a handful of recent studies, this effect could be widespread \citep{Shafer_Wolf_2013, Sexton_etal_2014}, although more detailed studies of IBE are now necessary to determine its full extent, the relative strengths of the various processes that underlie it, and the range of ecological conditions necessary for its generation.
	Isolation by environment is distinct from isolation by distance, which has been the foundation for examining gene flow and population structure for most of a century.  Hence, the study of IBE will provide new insights into mechanisms of dispersal, patterns of connectivity, and modes of differentiation both within and between species.  These will likely have major implications for a wide variety of disciplines, and they should be especially important for understanding how organisms will respond to rapid ecological change.  By incorporating environmental forces and ecological interactions – and potentially their spatial variation– into analyses of genetic divergence, studies of IBE present opportunities to identify the important factors that influence population, community, and even ecosystem dynamics in space.  Thus, the conceptual framework of IBE will play an important role in advancing our understanding of how ecology shapes the evolution of biological diversity.  

\section*{Acknowledgements}
	We thank S. Aeschbacher, Y. Brandvain, G. Coop, G. Gartner, A. Kai, J. Losos, L. Mahler, S. Spencer, and M. Weber for helpful advice and discussions during the preparation of this manuscript.


%\section*{Figure Legends}
%\subsection*{Fig. 1: Isolation by distance and environment.}
%Under the patterns of isolation by distance (IBD) and isolation by environment (IBE), genetic distance increases with geographic and environmental distance.  The three panels show different views of a simulated dataset in which both patterns can be seen.  Points represent the genetic distance (Gen. Dist.) between a pair of populations, plotted against their geographic (Geo. Dist.) and environmental distances (Env. Dist.), and are heat-colored by the magnitude of that environmental distance.

%\subsection*{Fig. 2: Illustration of processes that can generate a pattern of isolation by environment.}
% Dispersal between divergent environments can be reduced when (1) natural selection acts upon immigrants adapted to different environmental conditions, (2) sexual selection limits the reproductive success of immigrants with alternative traits, (3) hybrid offspring of native and immigrant parents have reduced fitness, for instance due to intermediate phenotypes, (4a) biased dispersal resulting from a genotype or phenotype leading to a dispersal preferences for particular environments, or (4b) biased dispersal resulting from a plastic natal habitat preference.

%\subsection*{Fig. 3: Illustration of common pitfalls in designing a sampling scheme.}
%These panels show the contrasted outcomes between complete sampling and an inadequately designed sampling scheme for populations on a hypothetical landscape with environmental variation (A).  The histogram of pairwise distances (B) obtained by a poor sample design shows that this design results in obtaining pairwise distances that are not fully representative of the full set of populations.  The comparison of the environmental and geographic distances recovered under the complete and partial sampling schemes (C) shows that these distances are highly correlated under the partial sampling scheme; IBD and IBE are conflated because all comparisons over short distances are also similar in environment, and all comparisons over long distances are environmentally disparate, with no intermediate comparisons to help disentangle the relative contributions of IBD and IBE.  Finally, pairwise genetic differentiation plotted against pairwise geographic distance and colored by environmental distance (D) shows that under the partial sampling scheme, the shape of the curves for IBD and IBE are poorly estimated.
